### PR TITLE
[40] Integrate form points with backend data

### PR DESCRIPTION
**Pull request by AnCichocka**
- make points form options use real data
- enable max points validation

image

image

### CODE REVIEW

File: frontend/src/components/form/PointsForm/PointsForm.tsx [hunk 1]
```diff
@@ -1,70 +1,80 @@
...
-      subcategoryId: values.subcategoryId,
-    };
-    handleAdd(points);
-  };
+type PointFormProps = {
+  handleAddPoints: (formPoints: FormPoints) => void;
+};
 
+export const PointsForm = ({ handleAddPoints: handleAdd }: PointFormProps) => {
   const formik = useFormik({
```
Comment (Szyntos) [inline frontend/src/components/form/PointsForm/PointsForm.tsx:22]: formik? xD

File: frontend/src/components/form/PointsForm/PointsForm.tsx [hunk 2]
```diff
@@ -1,70 +1,80 @@
...
     },
-    validate: validate,
-    onSubmit: onSubmit,
+    validate: (values: PointsFormValues) => {
+      try {
+        ValidationSchema.parse(values);
+      } catch (error) {
+        if (error instanceof ZodError) {
+          return error.formErrors.fieldErrors;
+        }
```
Comment (Szyntos) [inline frontend/src/components/form/PointsForm/PointsForm.tsx:34]: else?

**Review by Szyntos — APPROVED** [review]

File: frontend/src/components/form/PointsForm/PointsForm.tsx (hunk 1 above)
Comment (AnCichocka) [reply frontend/src/components/form/PointsForm/PointsForm.tsx:22]: that's how the library is named xd

File: frontend/src/components/form/PointsForm/PointsForm.tsx (hunk 2 above)
Comment (AnCichocka) [reply frontend/src/components/form/PointsForm/PointsForm.tsx:34]: there are no other errors, it's done only for proper error type mapping

**Pull request merged**